### PR TITLE
MULE-12060: Extension models for error handling

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 1]
```diff
@@ -1825,6 +1825,561 @@
...
+        "description": "",
+        "modelProperties": {}
+      },
+      "transactional": true,
+      "requiresConnection": false,
+      "parameterGroupModels": [
+        {
+          "parameters": [
+            {
+              "type": {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: this is the `transactionalAction` model that's generated for a transactional operation in DB:
`{
                  "type": {
                    "format": "java",
                    "type": "String",
                    "annotations": {
                      "typeId": "org.mule.runtime.extension.api.tx.OperationTransactionalAction",
                      "enum": {
                        "type": "[Ljava.lang.String;",
                        "values": [
                          "ALWAYS_JOIN",
                          "JOIN_IF_POSSIBLE",
                          "NOT_SUPPORTED"
                        ]
                      }
                    }
                  },
                  "hasDynamicType": false,
                  "required": false,
                  "expressionSupport": "NOT_SUPPORTED",
                  "defaultValue": "JOIN_IF_POSSIBLE",
                  "role": "BEHAVIOUR",
                  "dslConfiguration": {
                    "allowsInlineDefinition": true,
                    "allowsReferences": true,
                    "allowTopLevelDefinition": false
                  },
                  "layoutModel": {
                    "password": false,
                    "text": false,
                    "query": false,
                    "tabName": "Transactions"
                  },
                  "name": "transactionalAction",
                  "description": "The type of joining action that operations can take regarding transactions.",
                  "modelProperties": {}
                }`

Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: nice formatting... check `4.x-ce/extensions/db/target/classes/META-INF/extension-model.json`and look for `transactionalAction`

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 2]
```diff
@@ -1825,6 +1825,561 @@
...
+              "dslConfiguration": {
+                "allowsInlineDefinition": false,
+                "allowsReferences": false,
+                "allowTopLevelDefinition": false
+              },
+              "layoutModel": {
+                "password": false,
+                "text": false,
+                "query": false,
+                "order": 2
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "tabName": "Transactions"

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 3]
```diff
@@ -1825,6 +1825,561 @@
...
+              "dslConfiguration": {
+                "allowsInlineDefinition": false,
+                "allowsReferences": false,
+                "allowTopLevelDefinition": false
+              },
+              "layoutModel": {
+                "password": false,
+                "text": false,
+                "query": false,
+                "order": 1
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "tabName": "Transactions"

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 4]
```diff
@@ -1825,6 +1825,561 @@
...
+              },
+              "name": "name",
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: not necessary, once we support scopes as top-level-elements we should handle this as an special parameter.

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 5]
```diff
@@ -1825,6 +1825,561 @@
...
+        "description": "",
+        "modelProperties": {}
+      },
+      "transactional": false,
+      "requiresConnection": false,
+      "parameterGroupModels": [
+        {
+          "parameters": [
+            {
+              "type": {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: we should probably model this as an unionType of String + the enum ("REDELIVERY_EXHAUSTED", etc) so we don't lose support in tooling for the dropdown.

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 6]
```diff
@@ -1825,6 +1825,561 @@
...
+              "name": "when",
+              "description": "The expression that will be evaluated to determine if this exception strategy\nshould be executed. This should always be a boolean expression.",
+              "modelProperties": {}
+            },
+            {
+              "type": {
+                "format": "java",
+                "type": "Boolean",
+                "annotations": {
+                  "typeId": "boolean"
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "typeId": "java.lang.Boolean",

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 7]
```diff
@@ -1825,6 +1825,561 @@
...
+              "name": "logException",
+              "description": "Determines whether the handled exception will be logged to its standard logger in the\nERROR level before being handled. Default is true.",
+              "modelProperties": {}
+            },
+            {
+              "type": {
+                "format": "java",
+                "type": "Boolean",
+                "annotations": {
+                  "typeId": "boolean"
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "typeId": "java.lang.Boolean",

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 8]
```diff
@@ -1825,6 +1825,561 @@
...
+              },
+              "name": "logException",
+              "description": "Determines whether the handled exception will be logged to its standard logger in the\nERROR level before being handled. Default is true.",
+              "modelProperties": {}
+            },
+            {
+              "type": {
+                "format": "java",
+                "type": "Boolean",
+                "annotations": {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "annotations": {
                            "typeId": "java.lang.Boolean",
                            "default": "true"
                          }

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 9]
```diff
@@ -1825,6 +1825,561 @@
...
+              },
+              "name": "when",
+              "description": "The expression that will be evaluated to determine if this exception strategy\nshould be executed. This should always be a boolean expression.",
+              "modelProperties": {}
+            },
+            {
+              "type": {
+                "format": "java",
+                "type": "Boolean",
+                "annotations": {
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: "annotations": {
                            "typeId": "java.lang.Boolean",
                            "default": "true"
                          }

File: modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json [hunk 10]
```diff
@@ -1825,6 +1825,561 @@
       ],
       "description": "Processes the nested list of message processors asynchronously using a thread pool",
       "modelProperties": {}
+    },
+    {
+      "name": "try",
+      "kind": "scope",
+      "blocking": true,
+      "executionType": "CPU_LITE",
```
Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: BLOCKING

Comment (alegmarra) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: check org.mule.runtime.api.meta.model.ExecutionType

Comment (elrodro83) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: why?

Comment (elrodro83) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: why?

Comment (afelisatti) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: @dfeist can you check whether this makes sense? The try does build a chain with an intercepting processor that does not have a NB implementation. Can we be certain the other components are CPU_LITE?

Comment (afelisatti) [inline modules/extensions-support/src/main/resources/META-INF/mule-extension-model.json]: Just got confirmation this is ok, try should be BLOCKING and the rest are ok as CPU_LITE.

**Pull request merged**